### PR TITLE
Diffraction-related additions and fixes

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Trying to get closer to the point where we can properly load a diffraction calibration file and use it for unit conversion, and later diffraction-focussing.

- Some fixes in module setup.
- Remove duplicate file.
- Store detector->spectrum mapping when converting Mantid workspace to scipp.
- Improve structure of loaded calibration data.
- Ensure that calibration has correct dimensions, convert using det->spec mapping if found.

Not sure about all the details of the latter, may require some work, and *testing*. Let me know what you think so far.

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
Maybe it would be better to move the code for pre-processing the cal table out of `convert_with_calibration` and into a new function? Users would then need to call this by hand, i.e., deal with an grouping of detectors on their own.

File: neutron/diffraction/convert_with_calibration.cpp [hunk 1]
```diff
@@ -2,31 +2,56 @@
...
+    // 2b. Translate detector-based calibration information into coordinates of
+    // data.
+    // We are hard-coding some information here: the existence of "spectra",
+    // since we require labels named "spectrum" and a corresponding dimension.
+    // Given that this in in a branch that is access only if "detector_info" is
+    // present this should probably be ok.
+    cal = groupby(cal, "spectrum", Dim::Spectrum).mean(Dim::Detector);
+  } else if (d.dimensions().count(cal["tzero"].dims().inner()) != 1) {
+    throw except::DimensionError(
+        "Calibration depends on dimension that is not present in the converted "
```
Comment (OwenArnold) [inline neutron/diffraction/convert_with_calibration.cpp]: Doesn't report what that dimension is.

File: python/src/scipp/compat/mantid.py [hunk 2]
```diff
@@ -2,15 +2,15 @@
 # Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
 # @author Simon Heybrock, Neil Vaytet
 
-import scipp as sc
+from .._scipp import core as sc
```
Comment (OwenArnold) [inline python/src/scipp/compat/mantid.py]: Why this change `.._scipp` implies we are now intending to use it from CMAKE_BINARY directory directly?

File: python/src/scipp/compat/mantid.py [hunk 3]
```diff
@@ -26,8 +26,46 @@ def convert_instrument(ws):
...
+    nDet = det_info.size()
+    spectrum = sc.Variable([sc.Dim.Detector],
+                           shape=(nDet, ),
+                           dtype=sc.dtype.int32)
+    has_spectrum = sc.Variable([sc.Dim.Detector],
+                               values=np.full((nDet, ), False))
+    spectrum_ = spectrum.values
+    has_spectrum_ = has_spectrum.values
+    spec_info = ws.spectrumInfo()
+    for i in range(nHist):
```
Comment (OwenArnold) [inline python/src/scipp/compat/mantid.py]: Minor suggested improvement...

Iterators now supported. Should be able to do:
```
for spec in spec_info:
    ...
```

File: python/src/scipp/compat/mantid.py [hunk 4]
```diff
@@ -26,8 +26,46 @@ def convert_instrument(ws):
...
+            det, time = spec_def[j]
+            if time != 0:
+                raise RuntimeError(
+                    "Conversion of Mantid Workspace with scanning instrument "
+                    "not supported yet.")
+            spectrum_[det] = i
+            has_spectrum_[det] = True
+    detector = sc.Variable([sc.Dim.Detector], values=det_info.detectorIDs())
+
+    # Remove any information about detectors withou data (a spectrum). This
```
Comment (OwenArnold) [inline python/src/scipp/compat/mantid.py]: typo `withou`

File: .travis/check_clang_format.sh [hunk 5]
```diff
@@ -12,7 +12,7 @@ else
 fi
 
 # Perform formatting
-find core python test units -type f -name '*.h' -o -name '*.cpp' | xargs ${CLANG_FORMAT_EXE} -i -style=file;
+find common core neutron python test units -type f -name '*.h' -o -name '*.cpp' | xargs ${CLANG_FORMAT_EXE} -i -style=file;
```
[the remark this answers is not in the document]
Comment (OwenArnold) [reply .travis/check_clang_format.sh]: actually `git diff origin/master --name-only -- '*.cpp' '*.h'` is probably what you want

File: neutron/diffraction/convert_with_calibration.cpp [hunk 6]
```diff
@@ -2,31 +2,56 @@
...
-  for (const auto & [ name, data ] : d) {
+  // 5. Transform variables
+  for (const auto &[name, data] : d) {
     static_cast<void>(name);
     if (data.coords()[Dim::Tof].dims().sparse()) {
-      data.coords()[Dim::Tof].assign(
-          data.coords()[Dim::Tof] - cal["tzero"].data());
-      data.coords()[Dim::Tof].assign(
-          data.coords()[Dim::Tof] / cal["difc"].data());
+      data.coords()[Dim::Tof].assign(data.coords()[Dim::Tof] -
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply neutron/diffraction/convert_with_calibration.cpp]: `convert_with_calibration` does not do an in-place modification, so this should be irrelevant.

File: python/src/scipp/compat/mantid.py (hunk 2 above)
Comment (SimonHeybrock) [reply python/src/scipp/compat/mantid.py]: Previously `scipp` imported `neutron`, which imported `load` which imported `scipp`, so there was a cyclic dependency. Apparently Python managed to deal with it, but it did not seem like a good setup. It also appeared to cause some issues when trying to import submodules of `scipp` directly.

I don't know if this is the best solution, let me know if there is a better way.

File: core/include/scipp/core/variable.h [hunk 7]
```diff
@@ -652,6 +653,10 @@ class SCIPP_CORE_EXPORT VariableConstProxy {
   template <class T> auto sparseVariances() const {
     return castVariances<sparse_container<T>>();
   }
+  template <class T> const auto &value() const {
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply core/include/scipp/core/variable.h]: Thank you for pointing out my laziness (I mean it). In fact adding tests uncovered an (unrelated) bug.

File: python/src/scipp/compat/mantid.py (hunk 2 above)
Comment (OwenArnold) [reply python/src/scipp/compat/mantid.py]: Ok. Makes sense, and I don't have a better solution for handling that.
